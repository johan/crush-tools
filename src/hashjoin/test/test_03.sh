test_number=03
description="one-key join by index w/out headers"

# There are 2 records in the dimension file with a first field of 1.  The
# 2nd should override the first.

infile="$test_dir/input_no_header.log"
dimfile="$test_dir/dimension_no_header.log"
outfile="$test_dir/test_$test_number.actual"
expected="$test_dir/test_$test_number.expected"

$bin -k 1 -l 1 -j 3,4 -f $dimfile $infile \
  > "$outfile"

if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number 1 "$description" FAIL
else
  test_status $test_number 1 "$description" PASS
  rm "$outfile"
fi
