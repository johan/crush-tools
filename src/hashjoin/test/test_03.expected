1,2,wee,4,5
5,6,w00t,7,8
7,8,nope,,
